sources/packages/backend/libs/integrations/src/services/sfas/sims-to-sfas.service.ts

```diff
@@ -119,6 +124,123 @@ export class SIMSToSFASService {
     return modifiedStudentIds;
   }
 
+  /**
+   * Get all student ids and application data of students who have one or more updates in application related data.
+   * @param modifiedSince the date after which the application data was updated.
+   * @param modifiedUntil the date until which the application data was updated.
+   * @returns student ids and application data of students who have one or more updates in application related data.
+   */
+  async getAllApplicationsWithUpdates(
+    modifiedSince: Date,
+    modifiedUntil: Date,
+  ): Promise<ApplicationRecord[]> {
+    return (
+      this.applicationRepo
+        .createQueryBuilder("application")
+        .select("application.id", "applicationId")
+        .addSelect("student.id", "studentId")
+        .addSelect("programYear.programYear", "programYear")
+        .addSelect(
+          "COALESCE(offering.offeringIntensity, (application.data->>'howWillYouBeAttendingTheProgram')::sims.offering_intensity)",
+          "offeringIntensity",
+        )
+        // Use CASE to conditionally select studyStartDate and studyEndDate and casting JSON values to dates.
+        .addSelect(
+          "COALESCE(offering.studyStartDate, (application.data->>'studystartDate')::date)",
+          "studyStartDate",
+        )
+        .addSelect(
+          "COALESCE(offering.studyEndDate, (application.data->>'studyendDate')::date)",
+          "studyEndDate",
+        )
+        // Summing CSGP awards where value_code is 'CSGP'.
+        .addSelect(
+          "SUM(CASE WHEN disbursementValues.valueCode = 'CSGP' THEN disbursementValues.valueAmount ELSE 0 END)",
```

Comment:
As we discussed, sending award totals for cancelled applications doesn't seem to be right. Let's hear what business has to say.

Reply:
As discussed with @CarlyCotton , for now sending the awards for the cancelled applications.